docs/explanation/charm-architecture.md

```diff
@@ -0,0 +1,109 @@
+throughout the workload lifecycle.
+
+Pebble `services` are configured through [layers](https://github.com/canonical/pebble#layer-specification),
+and the following containers represent each one a layer forming the effective
+Pebble configuration, or `plan`:
+
+1. An [NGINX](https://www.nginx.com/) container, which can be used to
+efficiently serve static resources, as well as be the incoming point for all web
+traffic to the pod.
+2. The [Synapse](https://github.com/matrix-org/synapse) container itself, which
+has Synapse installed and configured.
+
+As a result, if you run a `kubectl get pods` on a namespace named for the Juju
+model you've deployed the Synapse charm into, you'll see something like the
+following:
+
+```bash
+NAME                             READY   STATUS    RESTARTS   AGE
+synapse-0                         3/3     Running   0         6h4m
+```
+
+This shows there are 3 containers - the two named above, as well as a container
+for the charm code itself.
+
+And if you run `kubectl describe pod synapse-0`, all the containers will have as
+Command ```/charm/bin/pebble```. That's because Pebble is responsible for the
+processes startup as explained above.
+
+## OCI images
+
+We use [Rockcraft](https://canonical-rockcraft.readthedocs-hosted.com/en/latest/)
+to build OCI Images for Synapse and NGINX.
+The images are defined in [NGINX ROCK](https://github.com/canonical/synapse-operator/tree/main/nginx_rock/)
+and [Synapse ROCK](https://github.com/canonical/synapse-operator/tree/main/synapse_rock).
+They are published to [Charmhub](https://charmhub.io/), the official repository
+of charms.
+This is done by publishing a resource to Charmhub as described in the
+[Juju SDK How-to guides](https://juju.is/docs/sdk/publishing).
+
+## Containers
+
+Configuration files for the containers can be found in the respective
+directories that define the ROCKs, see the section above.
+
+### NGINX
+
+This container is the entry point for all web traffic to the pod (on port
+`8080`). Serves some static files directly and forwards non-static requests to
+the Synapse container (on port `8008`).
+
+The reason for that is since NGINX provides cache static content, reverse proxy,
+and load balance among multiple application servers, as well as other features
+it can be used in front of Synapse server to significantly reduce server and
+network load.
+
+The workload that this container is running is defined in the [NGINX ROCK](https://github.com/canonical/synapse-operator/tree/main/nginx_rock/).
+
+### Synapse
+
+Synapse is a Python application run by the script "/start.py".
+
+Synapse listens by default in a no-TLS port `8008` serving it so NGINX can
+forward non-static traffic to it.
+
+The workload that this container is running is defined in the [Synapse ROCK](https://github.com/canonical/synapse-operator/tree/main/synapse_rock).
+
+## Integrations
+
+Please, see [Integrations](https://charmhub.io/synapse/docs/reference/integrations).
```

Comment:
I remember from one of the writing guides that "Please" is not preferred in documentation, to set the mood. Let me add the reference if I can find it!

Reply:
You are absolutely right, it's on the link that you shared
https://developers.google.com/style/tone#politeness-and-use-of-please
I'll remove it, thanks!